crates/math/src/fft/polynomial.rs

```diff
@@ -27,7 +27,9 @@ impl<E: IsField> Polynomial<FieldElement<E>> {
     ) -> Result<Vec<FieldElement<E>>, FFTError> {
         let domain_size = domain_size.unwrap_or(0);
         let len = core::cmp::max(poly.coeff_len(), domain_size).next_power_of_two() * blowup_factor;
-
+        if len.trailing_zeros() as u64 > F::TWO_ADICITY {
```

Comment:
also this is weird, having the `TWO_ADICITY` as u64 but then everything else is either `usize` or `u32`